Ta15775 backbone level improvements

LeopoldoFu opened the pull request:
@ebsco/eac-software-engineers Fun times.

LeopoldoFu committed:
Remove use of this._previousAttributes from set and changedAttributes.  Essentially side-steps _.cloning attributes.
Replace _.isEqual checks with strict equal checks.

LeopoldoFu committed:
Use _.extend instead of defaults in Backbone.Collection.set.

LeopoldoFu committed:
Remove unused functionality related to keeping track of previous attributes.
Update unit tests.

LeopoldoFu committed:
Fix model.changedAttributes(diff) so it still returns changes caused by a previous model.set().

Rogibb111 commented on backbone.js:
```diff
@@ -400,29 +399,16 @@
       return changed;
     },
 
-    // Get the previous value of an attribute, recorded at the time the last
```
good riddance

Rogibb111 commented on backbone.js:
```diff
@@ -400,29 +399,16 @@
     // You can also pass an attributes object to diff against the model,
     // determining if there *would be* a change.
     changedAttributes: function(diff) {
-      if (!diff) return this.hasChanged() ? _.clone(this.changed) : false;
```
no more clone!

LeopoldoFu replied:
![image](https://cloud.githubusercontent.com/assets/4174678/7731953/1300188e-fef2-11e4-91f8-7c796a7d12d5.png)

Rogibb111 commented on backbone.js:
```diff
@@ -341,8 +340,8 @@
       // For each `set` attribute, update or delete the current value.
       for (attr in attrs) {
         val = attrs[attr];
-        if (!_.isEqual(current[attr], val)) changes.push(attr);
```
boo underscore functions

Rogibb111 commented:
Looks good to me.

leachryanb commented:
You gonna submit a PR back to jashkenas?

LeopoldoFu commented:
@leachryanb no, We are truly forking backbone here;  we've changed how backbone handles diffing model changes from deep comparison to simple === comparison and also removed the functionality to get the previous model attributes at any time.

leachryanb commented:
Huh, wonder if we should change the name to eis-backbone or something.

On Wed, May 20, 2015 at 2:59 PM, LeopoldoFu notifications@github.com
wrote:

> @leachryanb https://github.com/leachryanb no, We are truly forking
> backbone here; we've changed how backbone handles diffing model changes
> from deep comparison to simple === comparison and also removed the
> functionality to get the previous model attributes at any time.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ebsco/backbone/pull/2#issuecomment-103996079.

## 

-rbl
